### ISSUE TITLE
RCS Support

### ISSUE THREAD

**Issue by aaronasmith** [klinker-apps/messenger-issues#143]
Google messenger has started to activate it for some people and I'd like to be able to use it with your app.

**Comment by klinker24** [comment]
I have a FAQ about this:

```
This is something I would love to have, however, it may take time without Google's support. They bought Jibe (a company focused on bringing RCS to the masses) in Sept. 2015. It still took them well over a year to roll out support after that multi-million dollar buyout. I don't have those funds, so without support from Google (which I am hoping they will provide), it isn't realistic to think that RCS will come quickly to any third party applications.
```

Until we have more information, this won't be possible. Any developer can add support here if they think they know what they are doing: https://github.com/klinker41/android-smsmms

**Comment by klinker24** [comment]
Here is the protocol standard, stating that API support will come in "Release 2", which is planned for Q2 2017.

http://www.gsma.com/network2020/universal-profile/

**Comment by dvranizan** [comment]
Release 2 is now available.  I browsed it.  Annex B is particularly interesting in regards to getting the RCS stack up and going in Android.

I'm going to look into getting this into android-smsmms .

[RCC.71_v2.0.pdf](https://github.com/klinker-apps/messenger-issues/files/1149272/RCC.71_v2.0.pdf)

**Comment by klinker41** [comment]
Best of luck, that is an extremely lengthy document! Let me know if you have any questions surrounding working with the library. It's been quite a while since I've put a lot of time into it, but might be able to answer questions if they come up.

**Comment by Orion116** [comment]
I recently came across this: https://github.com/android-rcs/rcsjta

**Comment by danielling23** [comment]
Huawei has just committed to supporting Google's implementation of RCS for all their phones. Hopefully this will get more support moving forward. 

[https://9to5google.com/2018/01/18/huawei-preload-android-messages-google-partnership/](url)
[https://venturebeat.com/2018/01/18/google-and-huawei-partner-to-bring-rcs-to-millions-via-jibe-and-android-messages-app/](url)
[http://www.androidpolice.com/2018/01/17/huawei-endorses-google-jibe-rcs-will-start-including-android-messages-phones/](url)
[http://consumer.huawei.com/en/press/news/2018/google-huawei-form-strategic-partnership-on-rcs-messaging/](url)

**Comment by scrumit** [comment]
The standard document has this quote at the beginning.

> The focus of requirements is on native device 
implementations implemented at the OS or device manufacturer level. It is acknowledged 
that downloadable applications may face limitations that prevent such implementations from 
fulfilling the complete feature set.

The "may" will almost certainly mean "will".  That is, while pulse is just an app. You may be able to get to the communications firmware if you are a part of fuchsia.

**Comment by xEuPhoRiA-** [comment]
Once RCS is available on my carrier, sadly I will have to move away from Pulse. Current SMS and MMS compression on images and videos is crazy when I send using Pulse. Most people I send images to using Pulse tell me they are virtually unreadable or watchable when they receive them. I would love to continue supporting Pulse but RCS is a huge improvement and beings more value that Pulse would at that time. Specially with the recent findings from xda that Android Messages looks to be adding WiFi sending support as well as the possibility of a browser version. Any comment on updated details about RCS in Pulse?

**Comment by klinker24** [comment]
You can see all the comments here already, discussing that RCS support will require integration support from Google. It isn't something that apps will be able to do on their own. Google has been building the backend for their RCS support for 3-4 years. It isn't something that I have the ability to build, for Pulse.

Without integration from Google, RCS will likely never come to Pulse. You will get the same response on this from any third party app developer, unfortunately.

You can always just share a link to your videos. Google Photos is awesome.

**Comment by xTcoops** [comment]
@klinker24  I totally understand and indeed read the comments. I just hadn't seen a comment in a while so wanted to get a refreshed look at the issue and see if the limitations were still there. It seems they are, which is unfortunate.  Thanks for the speedy reply, and I will continue to support Pulse until I can no longer, I was happy to pay the nominal fee for the lifetime subscription, was even willing to pay more to support the little guys. Keep it up, I hope to see improvement on Google's side to allow apps to be able to use RCS as well.

**Comment by jadias1** [comment]
"Carrier Services" is an app that Google is pushing to Android platforms to increase adoption of RCS.  Not sure if that everything is there (libs & dependencies, etc...) that might help you with Pulse SMS.  Didn't seem to be listed before so I thought I'd mention it.  

https://play.google.com/store/apps/details?id=com.google.android.ims&hl=en 

Best,
J

**Comment by klinker24** [comment]
No, that isn't really the type of thing that is going to be helpful to Pulse.

**Comment by rje6459** [comment]
Verizon just begun rolling out support for RCS on the Pixel 3 and 3XL. Any updates for RCS support with Pulse?

**Comment by rnhurt** [comment]
The folks over at [r/UniversalProfile](https://www.reddit.com/r/UniversalProfile) have put together a [spreadsheet with RCS support details](https://docs.google.com/spreadsheets/d/14Ns9oV0Dh8_S5M-JHDvJ8sNHR6rpbiq2N6qZlP3g9zQ/edit#gid=0).  It would be nice to get PulseSMS on this sheet, probably listed as "Waiting for APIs".

**Comment by Golgautier** [comment]
Hi all. RCS are supported by message app from Google in France & UK. Google servers are used instead of carrier's ones (Google hopes impulse RCS adoption by doing this).

Do you have news about support by Pulse ?

**Comment by klinker24** [comment]
> As noted above, until Google adds something to Android, it won't be available for third party apps. I will post something myself, if this changes. A new carrier supporting RCS has no implications for third party apps.

I have no updates, nothing has changed, I would post or update the issue if something changed.

**Comment by RacingJayson** [comment]
@rnhurt Done.

**Comment by janssjay** [comment]
https://www.androidcentral.com/want-rcs-chat-your-phone-now-heres-how-enable-it

**Comment by RacingJayson** [comment]
@janssjay This has nothing to do with RCS Support for third-party apps.

**Comment by bertradio** [comment]
Just saw a news item that Google is rolling out RCS. 

Anything new on this?

**Comment by klinker24** [comment]
No, that does not mean anything for third party apps.

**Comment by RacingJayson** [comment]
@bertradio This has no effect on third-party app support.

**Comment by bertradio** [comment]
Understood. Any indication as to when that may happen?

If someone uses Messages to send me a text RCS and Pulse is my default on my phone, will I get it? It is my understanding that Messages checks as to whether the recipient has RCS and if not, it sends via normal SMS. I do not want to miss any messages.

**Comment by klinker24** [comment]
You just need to disable RCS on your phone.

https://messenger.klinkerapps.com/help

I have no updates to share on RCS and will post here if I do have an update. Google has given no indication on third party app support, as we have discussed above.

**Comment by klinker24** [comment]
I have locked this topic and will unlock it for further discussion and implementation when/if something changes around support for third party apps and RCS.
